GitHub_Learning/Feature/Dashboard/Resource/CryptocurrencyParser.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  CryptocurrencyPriceParser.swift
+//  GitHub_Learning
+//
+//  Created by Amin Ashhab on 20.12.2020.
+//  Copyright © 2020 Amin_Second_Test_Project. All rights reserved.
+//
+
+import Foundation
+import CleanCore
+
+protocol CryptocurrencyParser {
+    func parse(_ responseBody: DeserializedBody) throws -> Cryptocurrency
+}
+
+class CryptocurrencyParserImpl: Parser, CryptocurrencyParser {
+    func parse(_ responseBody: DeserializedBody) throws -> Cryptocurrency {
+        self.deserializedBody = responseBody
+        let array = deserializedBody as? [Any]
+        let dictionary = array?[0] as? [String:Any]
```

Comment:
```suggestion
        let dictionary = array?.first as? [String: Any]
```
V tomto pripade je bezpecnejsi pouzit first. Pokud by tam totiz ten prvni prvek nebyl, tak by ti to ted spadlo. S tim first ti to jen vrati nil.